mindmaps-engine/src/main/java/io/mindmaps/engine/backgroundtasks/AbstractTaskManager.java

```diff
@@ -0,0 +1,181 @@
+                if (status == RUNNING && (!future.isDone()))
+                    state.setPauseState(future.get().pause());
+
+                future.cancel(true);
+            } catch (InterruptedException | ExecutionException e) {
+                LOG.error(e.getMessage());
+            }
+
+            updateTaskState(uuid, state);
+        } else {
+            LOG.warn("Ignoring pause request from: "+Thread.currentThread().getStackTrace()[2].toString()+
+                     " for task "+uuid.toString()+
+                     " because its status is "+state.getStatus());
+        }
+
+        return this;
+    }
+
+    public TaskManager resumeTask(UUID uuid, String requesterName, String message) {
+        TaskState state = getTaskState(uuid);
+        if(state.getStatus() != PAUSED) {
+           LOG.warn("Ignoring resume request from: "+Thread.currentThread().getStackTrace()[2].toString()+
+                     " for task "+uuid.toString()+
+                     " because its status is "+state.getStatus());
+            return this;
+        }
+
+        state.setStatus(RUNNING)
+            .setStatusChangeMessage(message)
+            .setStatusChangedBy(requesterName);
+        uuid = updateTaskState(uuid, state);
+
+        try {
+            BackgroundTask task = instantiateTask(state.getName());
+            task.resume(state.getPauseState());
+
+            if(state.getRecurring())
+                executeRecurring(uuid, task, state.getDelay(), state.getInterval());
+            else
+                executeSingle(uuid, task, state.getDelay());
+
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            LOG.error("Unable to resume task "+uuid.toString()+" the error was: "+e.getMessage());
+        }
+
+        return this;
+    }
+
+    public TaskManager restartTask(UUID uuid, String requesterName, String message) {
+        TaskState state = getTaskState(uuid);
+        TaskStatus status = state.getStatus();
+        if(status == SCHEDULED || status == CREATED || status == RUNNING) {
+            LOG.warn(Thread.currentThread().getStackTrace()[2].toString()+" tried to call restartTask() on a "+status.toString()+" task. IGNORING.");
+            return this;
+        }
+
+        try {
+            // Clean up etc
+            BackgroundTask task = instantiateTask(state.getName());
+            task.restart();
+
+            // Start it up again
+            if(state.getRecurring())
+                executeRecurring(uuid, task, state.getDelay(), state.getInterval());
+            else
+                executeSingle(uuid, task, state.getDelay());
+
+            // Update status message
+            state.setStatus(SCHEDULED)
+                 .setStatusChangeMessage(message)
+                 .setStatusChangedBy(requesterName);
+            updateTaskState(uuid, state);
+        } catch(ClassNotFoundException | InstantiationException | IllegalAccessException e) {
```

Comment:
I see these catch clauses everywhere. Is there no way to handle them in the base method `instantiateTask` ?